### ISSUE TITLE
Cohere Family of Models (Command-R, Command-R-Plus, Aya23-8B, Aya23-35B, Aya101)

### ISSUE THREAD

**Issue by user-0a** [NVIDIA/TensorRT-LLM#1657]
### System Info

Hello,

I am creating this issue for the purpose of requesting support for the Cohere family of models:

Command-R: https://huggingface.co/CohereForAI/c4ai-command-r-v01
Command-R-Plus: https://huggingface.co/CohereForAI/c4ai-command-r-plus
Aya23-8B: https://huggingface.co/CohereForAI/aya-23-8B
Aya23-35B: https://huggingface.co/CohereForAI/aya-23-35B
Aya101: https://huggingface.co/CohereForAI/aya-101

Thank you

### Information

- [ ] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

-

### Expected behavior

-

### actual behavior

-

### additional notes

-

**Comment by imnoahcook** [comment]
I would also like to request support for CommandR and CommandR+ plus, they are currently the best open source models.

**Comment by aikitoria** [comment]
Yes please. Command-R+ support is needed!

**Comment by here4dadata** [comment]
+1

**Comment by syuoni** [comment]
Hi all, we've started investigating and implementing the cohere models. It is planned to be delivered in the 0.13 version.

**Comment by aikitoria** [comment]
Nice!

**Comment by aikitoria** [comment]
@syuoni Please also support these!
https://huggingface.co/CohereForAI/c4ai-command-r-08-2024
https://huggingface.co/CohereForAI/c4ai-command-r-plus-08-2024

**Comment by syuoni** [comment]
> [@syuoni](https://github.com/syuoni) Please also support these! https://huggingface.co/CohereForAI/c4ai-command-r-08-2024 https://huggingface.co/CohereForAI/c4ai-command-r-plus-08-2024

These models share the same architecture `CohereForCausalLM`, so it's very likely they will be supported automatically once command-r is ready.

**Comment by user-0a** [comment]
Thank you!

**Comment by salaki** [comment]
Who knows how Cohere Family of Models are similar to other OSS model supported in TensorRTLLM?

**Comment by salaki** [comment]
@syuoni is the feature delayed? If I wanted to create a convertor from huggingface definition to tensorrtllm checkpoint myself, what document I should check?

**Comment by aikitoria** [comment]
It doesn't look like it is out, none of the recent commits mention it, but the version 0.13 was just released...

**Comment by syuoni** [comment]
Hi all,

Yes. Cohere is postponed; it's not available in 0.13. Cohere has some structures quite different from LLaMA, e.g., qk_layernorm, so it took some extra time to align the accuracy.

The MR is ready and under review in our internal repo. I think it will be released soon. Thanks!

**Comment by syuoni** [comment]
Hi all, 

The Command-R and Aya models have been supported on main branch. See: https://github.com/NVIDIA/TensorRT-LLM/tree/main/examples/commandr

I'm closing this issue. Thanks!
